packages/nimble-components/src/table/components/row/index.ts

```diff
@@ -201,12 +201,14 @@ export class TableRow<
     }
 
     public onRowExpandToggle(event: Event): void {
-        const expandEventDetail: TableRowExpandToggleEventDetail = {
-            oldState: this.expanded,
-            newState: !this.expanded,
-            recordId: this.recordId!
-        };
-        this.$emit('row-expand-toggle', expandEventDetail);
+        if (typeof this.recordId === 'string') {
+            const expandEventDetail: TableRowExpansionToggleEventDetail = {
+                oldState: this.expanded,
```

Comment:
In a very quick scan, I didn't see anything setting `this.expanded = !this.expanded`. Can you make sure something is doing that and that we have test coverage to verify the event data for both expand and collapse?